pkg/aws/actuator/actuator.go

```diff
@@ -545,12 +547,12 @@ func (a *AWSActuator) syncMint(ctx context.Context, cr *minterv1.CredentialsRequ
 	}
 
 	// Check if the user has the expected tags:
-	if !userHasExpectedTags(logger, userOut, infraName, string(clusterUUID)) {
+	if !userHasExpectedTags(logger, userOut, string(clusterUUID), infra) {
 		if rootAWSClient == nil {
 			return fmt.Errorf("no root AWS client available, cred secret may not exist: %s/%s", constants.CloudCredSecretNamespace, constants.AWSCloudCredSecretName)
 		}
 
-		err = a.tagUser(logger, rootAWSClient, awsStatus.User, infraName, string(clusterUUID))
+		err = a.tagUser(logger, rootAWSClient, awsStatus.User, string(clusterUUID), infra)
```

Comment:
Will this work when certain resource tags are removed from Infrastructure resource. Do we expect them to be removed from cloud resources too?

Reply:
According to this [comment](https://github.com/openshift/enhancements/pull/1051/files#r821548397), the deletion of the tag from the Infra resource doesn't actually need to delete the tag from the existing AWS resource. It remains in the unman-aged state. Moreover, user needs to pass the empty string against any tag key that need to be deleted from the AWS resource.